sql/2018-04-17-subscribers-table.sql

```diff
@@ -0,0 +1,12 @@
+-- This file creates a new table, subscription, in the ejabberd DB.
+
+USE ejabberd;
+
+CREATE TABLE subscription (
+    jid varchar(50) NOT NULL,
```

Comment:
Should we have a unique constraint on jid/room or do you expect there to be dups intentionally?

Reply:
Dups are intentional.  A user can have more than one subscription

Reply:
To the same room?

Reply:
(I'm suggesting a unique constraint on `jid & room` although now that I'm thinking it, I wonder if that's a performance concern given that these are strings)

Reply:
Here is the query that I'm using

```        ejabberd_sql:sql_query_t(
                  ?SQL("insert into subscription (jid, room) "
                       "select %(SJID)s, %(SRoomJID)s from DUAL "
                       "where not exists ("
                       "   select 1 from subscription "
                       "   where jid=%(SJID)s AND room=%(SRoomJID)s);"))```

Example:
user 9 and 10 make room 9-10
user 9 and 11 make room 9-11

The db will have
jid      | room
9        | 9-10
10      | 9-10
9        | 9-11
11       | 9-11